core/container/src/main/java/org/wildfly/swarm/container/runtime/ConfigurableManager.java

```diff
@@ -0,0 +1,456 @@
+package org.wildfly.swarm.container.runtime;
+
+import java.lang.invoke.LambdaMetafactory;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.jboss.logging.Logger;
+import org.wildfly.swarm.config.runtime.Keyed;
+import org.wildfly.swarm.config.runtime.SubresourceInfo;
+import org.wildfly.swarm.internal.SwarmMessages;
+import org.wildfly.swarm.spi.api.Defaultable;
+import org.wildfly.swarm.spi.api.Fraction;
+import org.wildfly.swarm.spi.api.StageConfig;
+import org.wildfly.swarm.spi.api.annotations.Configurable;
+
+/**
+ * @author Bob McWhirter
+ */
+public class ConfigurableManager {
+
+    private static final Set<String> BLACKLISTED_FIELDS = new HashSet<String>() {{
+        add("pcs");
+        add("key");
+        add("subresources");
+    }};
+
+    private static final Set<Class<?>> BLACKLISTED_CLASSES = new HashSet<Class<?>>() {{
+        add(Map.class);
+        add(Properties.class);
+    }};
+
+    private static final Set<Class<?>> CONFIGURABLE_VALUE_TYPES = new HashSet<Class<?>>() {{
+        add(Boolean.class);
+        add(Boolean.TYPE);
+        add(Short.class);
+        add(Short.TYPE);
+        add(Integer.class);
+        add(Integer.TYPE);
+        add(Long.class);
+        add(Long.TYPE);
+        add(Float.class);
+        add(Float.TYPE);
+        add(String.class);
+
+        add(Defaultable.class);
+    }};
+
+    private static Logger LOG = Logger.getLogger("org.wildfly.swarm.config");
+
+    private final List<ConfigurableHandle> configurables = new ArrayList<>();
```

Comment:
I assume the list of `configurables ` can become quiet big, so I am wondering if it needs to be released at some point during the server bootstrap, i.e. when when the list bootstrap DMR operations has been build?